src/models/two_nodes.js

```diff
@@ -0,0 +1,886 @@
+ * @param {number[]} body_surface_area Body surface area, default value 1.8258 [m2] in [ft2] if units = ‘IP’
+ * @param {number[]} p_atmospheric Atmospheric pressure, default value 101325 [Pa] in [atm] if units = ‘IP’
+ * @param {"standing" | "sitting"} body_position Select either “sitting” or “standing”
+ * @param {number[]} max_skin_blood_flow Maximum blood flow from the core to the skin, [kg/h/m2] default 80
+ * @param {TwoNodesKwargs} [kwargs]
+ *
+ * @returns {TwoNodesArrayReturnType} object with results of two_nodes_array
+ *
+ * @example
+ * const results = two_nodes_array([25,30], [25,35], [0.3,0.5], [50,60], [1.2,1.5], [0.5, 0.3], [0], [1.8258], [101325], ["standing"], [90])
+ * console.log(results); // {
+  e_skin: [ 16.2, 60.4 ],
+  e_rsw: [ 7, 51.9 ],
+  e_max: [ 159.9, 193.8 ],
+  q_sensible: [ 47.6, 21.1 ],
+  q_skin: [ 63.8, 81.5 ],
+  q_res: [ 5.2, 5 ],
+  t_core: [ 36.9, 37 ],
+  t_skin: [ 33.7, 35.1 ],
+  m_bl: [ 12.9, 31.6 ],
+  m_rsw: [ 10.3, 76.3 ],
+  w: [ 0.1, 0.3 ],
+  w_max: [ 0.6, 0.6 ],
+  set: [ 23.6, 29.3 ],
+  et: [ 25, 32.5 ],
+  pmv_gagge: [ 0.1, 1.6 ],
+  pmv_set: [ -0, 1.1 ],
+  disc: [ 0.1, 1.9 ],
+  t_sens: [ 0.1, 1.4 ]
+}
+ *
+ */
+export function two_nodes_array(
+  tdbArray,
+  trArray,
+  vArray,
+  rhArray,
+  metArray,
+  cloArray,
+  wme = [0],
+  body_surface_area = [1.8258],
+  p_atmospheric = [101325],
+  body_position = ["standing"],
+  max_skin_blood_flow = [90],
+  kwargs = {},
+) {
+  const defaults_kwargs = {
+    calculate_ce: false,
+    round: true,
+    max_sweating: 500,
+    w_max: false,
+  };
+
+  let joint_kwargs = Object.assign(defaults_kwargs, kwargs);
+
+  const vaporPressureArray = rhArray.map((_rh, index) =>
+    cal_vapor_pressure(tdbArray[index], _rh),
+  );
+  const wmeArray = fill_array(wme, tdbArray);
+  const bodySurfaceArray = fill_array(body_surface_area, tdbArray);
+  const pAtmArray = fill_array(p_atmospheric, tdbArray);
+  const bodyPositionArray = fill_array(body_position, tdbArray);
+  const maxSkinBloodFlowArray = fill_array(max_skin_blood_flow, tdbArray);
+
+  const setArray = [];
+  const eSkinArray = [];
+  const eRswArray = [];
+  const eMaxArray = [];
+  const qSensArray = [];
+  const qSkinArray = [];
+  const qResArray = [];
+  const tCoreArray = [];
+  const tSkinArray = [];
+  const mBlArray = [];
+  const mRswArray = [];
+  const wArray = [];
+  const wMaxArray = [];
+  const etArray = [];
+  const pmvGaggeArray = [];
+  const pmvSetArray = [];
+  const discArray = [];
+  const tSensArray = [];
+
+  for (let index = 0; index < tdbArray.length; index++) {
+    const result = calculate_two_nodes(
+      tdbArray[index],
+      trArray[index],
+      vArray[index],
+      metArray[index],
+      cloArray[index],
+      vaporPressureArray[index],
+      wmeArray[index],
+      bodySurfaceArray[index],
+      pAtmArray[index],
+      bodyPositionArray[index],
+      joint_kwargs,
+    );
+    setArray.push(result.set);
```

Comment:
see the previous comment, if you allocate the arrays before hand you can just set the values to the index here, avoiding pushing and causing the array to reallocate and copy values over